homeassistant/components/meater/sensor.py

```diff
@@ -0,0 +1,175 @@
+"""The Meater Temperature Probe integration."""
+from datetime import timedelta
+from enum import Enum
+import logging
+
+import async_timeout
+from meater import AuthenticationError, TooManyRequestsError
+
+from homeassistant.const import DEVICE_CLASS_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.device_registry as dr
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Meater Temperature Probe sensor."""
+    return True
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the entry."""
+    # assuming API object stored here by __init__.py
+    api = hass.data[DOMAIN][entry.entry_id]
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            # Note: asyncio.TimeoutError and aiohttp.ClientError are already
+            # handled by the data update coordinator.
+            async with async_timeout.timeout(10):
+                devices = await api.get_all_devices()
+        except AuthenticationError as err:
+            raise UpdateFailed("The API call wasn't authenticated") from err
+        except TooManyRequestsError as err:
+            raise UpdateFailed(
+                "Too many requests have been made to the API, rate limiting is in place"
+            ) from err
+        except Exception as err:  # pylint: disable=broad-except
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+        # Populate the entities
+        entities = []
+
+        for dev in devices:
+            if dev.id not in hass.data[DOMAIN]["entities"]:
+                entities.append(
+                    MeaterProbeTemperature(
+                        coordinator, dev.id, TemperatureMeasurement.Internal
+                    )
+                )
+                entities.append(
+                    MeaterProbeTemperature(
+                        coordinator, dev.id, TemperatureMeasurement.Ambient
+                    )
+                )
+                device_registry = await dr.async_get_registry(hass)
+                device_registry.async_get_or_create(
+                    config_entry_id=entry.entry_id,
+                    identifiers={(DOMAIN, dev.id)},
+                    name=f"Meater Probe {dev.id}",
+                    manufacturer="Apption Labs",
+                    model="Meater Probe",
+                )
+                hass.data[DOMAIN]["entities"][dev.id] = None
```

Comment:
Why are we keeping track of the entities?

Reply:
It's to keep track of which temperature probes we have created entities for

Reply:
Sure, but why would that be needed? The coordinator handles the signaling?